### PR TITLE
bt-146: Create models for entities user details and hard skills from bt-72

### LINKED ISSUE #146: Create models and entities for migrations from bt-72
Create models and entities migrations from bt-72
- [x] create models
- [x] create entities

### PR DESCRIPTION
**Pull request by lavandosya**
I've added types for select components and change id type from number to string, also created models and entities for hard_skills and user_details.

### CODE REVIEW

**Review by MaksymGavatuk — APPROVED** [review]

File: backend/src/bundles/users/user.model.ts [hunk 1]
```diff
@@ -13,6 +15,17 @@ class UserModel extends AbstractModel {
     public static override get tableName(): string {
         return DatabaseTableName.USERS;
     }
+
+    public static override relationMappings = {
+        details: {
+            relation: Model.HasOneRelation,
+            modelClass: UserModel,
+            join: {
+                from: 'users.is',
```
Comment (nika-sirak) [inline backend/src/bundles/users/user.model.ts]: `from: 'users.id',`

File: backend/src/bundles/users/user.model.ts [hunk 2]
```diff
@@ -13,6 +15,17 @@ class UserModel extends AbstractModel {
...
+                to: 'users_details.user_id',
```
Comment (nika-sirak) [inline backend/src/bundles/users/user.model.ts]: user_details.user_id or user_details.userId?

**Review by nika-sirak — CHANGES_REQUESTED** [review]

File: backend/src/bundles/users/user.model.ts (hunk 1 above)
Comment (nikita-remeslov) [reply backend/src/bundles/users/user.model.ts]: not resolved

File: shared/src/bundles/users/types/employment-type.type.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+type EmploymentType =
+    | 'Full-Time'
```
Comment (nikita-remeslov) [inline shared/src/bundles/users/types/employment-type.type.ts]:
```suggestion
    | 'Full Time'
```

File: shared/src/bundles/users/types/employment-type.type.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+    | 'Part time'
+    | 'Freelance(projects)'
```
Comment (nikita-remeslov) [inline shared/src/bundles/users/types/employment-type.type.ts]:
```suggestion
    | 'Freelance/Projects'
```

File: shared/src/bundles/users/types/types.ts [hunk 5]
```diff
@@ -1,3 +1,9 @@
+export { type CountryList } from './country-list.type.js';
+export { type EmploymentType } from './employment-type.type.js';
+export { type EnglishLevel } from './english-level.type.js';
+export { type JobTitle } from './job-title.type.js';
+export { type NotConsidered } from './not-considered.type.js';
+export { type PreferredLanguages } from './preferred-languages.type.js';
```
Comment (nikita-remeslov) [inline shared/src/bundles/users/types/types.ts]: better make it like enums

File: backend/src/bundles/hard-skills/hard-skills.model.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        return DatabaseTableName.TALENT_HARD_SKILLS;
+    }
+
+    public static override relationMappings = {
+        talents: {
+            relation: Model.ManyToManyRelation,
+            modelClass: UserDetailsModel,
+            join: {
+                from: 'hard_skills.id',
+                through: {
```
Comment (nikita-remeslov) [inline backend/src/bundles/hard-skills/hard-skills.model.ts:25]: I think we should use one style.
`hard_skill_id`

Comment (nikita-remeslov) [reply backend/src/bundles/hard-skills/hard-skills.model.ts:25]: Also you should make it like this
```js
            join: {
                from: `${DatabaseTableName.USERS}.id`,
                to: `${DatabaseTableName.USER_DETAILS}.user_id`,
            },
```

Comment (nikita-remeslov) [reply backend/src/bundles/hard-skills/hard-skills.model.ts:25]: we should snake_case everywhere

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: backend/src/bundles/hard-skills/hard-skills.model.ts (hunk 6 above)
Comment (lavandosya) [reply backend/src/bundles/hard-skills/hard-skills.model.ts:25]: But in models we use camel case, that's why here it in camel case

File: backend/src/bundles/users/user.model.ts (hunk 1 above)
Comment (lavandosya) [reply backend/src/bundles/users/user.model.ts]: I resolved it in next commit

File: backend/src/bundles/hard-skills/hard-skills.model.ts (hunk 6 above)
Comment (nikita-remeslov) [reply backend/src/bundles/hard-skills/hard-skills.model.ts:25]: than for tables let use snake case and for column camel case

ok?

File: backend/src/bundles/hard-skills/hard-skills.model.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        talents: {
+            relation: Model.ManyToManyRelation,
+            modelClass: UserDetailsModel,
+            join: {
+                from: `${DatabaseTableName.HARD_SKILLS}.id`,
+                through: {
+                    from: `${DatabaseTableName.TALENT_HARD_SKILLS}.hardSkillsId`,
+                    to: `${DatabaseTableName.TALENT_HARD_SKILLS}.userDetailsId`,
+                },
+                to: 'user_details.id',
```
Comment (MaksymGavatuk) [inline backend/src/bundles/hard-skills/hard-skills.model.ts]: Use DatabaseTableName enum

**Review by MaksymGavatuk — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

**Pull request merged**